Dockerfile

```diff
@@ -0,0 +1,103 @@
+# Copyright 2019 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Do not add forseti source in this file to avoid rebuilding base during development.
+
+##### BEGIN BASE IMAGE #####
+FROM python:3.6-slim as base
+
+ENV HOME=/home/forseti
+
+RUN groupadd -g 1000 forseti && \
+    useradd -d ${HOME} -u 1000 -g forseti forseti && \
+    mkdir -p ${HOME}/forseti-security /forseti-security && \
+    chown -R forseti:forseti ${HOME} /forseti-security
+
+# Install host dependencies.
+RUN apt-get update  && \
+    apt-get install -y libmariadbclient18 && \
+    rm -rf /var/lib/apt/lists/*
+
+USER forseti
+##### END BASE IMAGE #####
+
+##### BEGIN PRE-BUILD IMAGE #####
+FROM base AS pre-build
+
+WORKDIR /forseti-security/
+
+USER root
+# Install Forseti Security dependencies.
+# This should stay in sync with the deployment script used on the host machine in
+#   deployment-templates/compute-engine/forseti-instance.py
+RUN apt-get update  && \
```

Comment:
Why do we run this twice, here and on line 28?

Reply:
We do this twice because there are actually two different container images being built.  The first is a "base" container.  It creates the foundation, sets up the user, an installs the minimum necessary packages for run time.

The "pre-build" stage adds host dependencies only necessary for compile and test.